dev-libs/openssl/openssl-1.0.1r.recipe

```diff
@@ -11,26 +10,23 @@ OpenSSL is based on the excellent SSLeay library developed by Eric A. Young \
 and Tim J. Hudson. The OpenSSL toolkit is licensed under an Apache-style \
 licence, which basically means that you are free to get and use it for \
 commercial and non-commercial purposes subject to some simple license \
-conditions.
-"
-HOMEPAGE="http://www.openssl.org/"
-SOURCE_URI="http://www.openssl.org/source/openssl-${portVersion}.tar.gz"
-CHECKSUM_SHA256="9d1c8a9836aa63e2c6adb684186cbd4371c9e9dcc01d6e3bb447abf2d4d3d093"
+conditions."
+HOMEPAGE="https://www.openssl.org/"
+COPYRIGHT="1995-1998 Eric Young
+	1998-2016 The OpenSSL Project."
 LICENSE="OpenSSL"
-COPYRIGHT="
-	1995-1998 Eric Young
-	1998-2013 The OpenSSL Project.
-	"
 REVISION="1"
-ARCHITECTURES="?x86_gcc2 x86 ?x86_64"
-SECONDARY_ARCHITECTURES="?x86_gcc2 ?x86"
+SOURCE_URI="https://www.openssl.org/source/openssl-$portVersion.tar.gz"
+CHECKSUM_SHA256="784bd8d355ed01ce98b812f873f8b2313da61df7c7b5677fcf2e57b0863a3346"
+PATCHES="openssl-$portVersion.patchset"
 
-PATCHES="openssl-1.0.1h.patchset"
+ARCHITECTURES="x86_gcc2 x86 ?x86_64 ?arm ?ppc"
+SECONDARY_ARCHITECTURES="x86_gcc2 ?x86 ?x86_64 ?arm ?ppc"
```

Comment:
Maybe the build creates a dummy self signed certificate, and on cross builds it needs a native build. This is just a guess because I haven't digged the question.